third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/WSCodePartition.cpp

```diff
@@ -1523,8 +1523,10 @@ desyncTCGen5MMAOp(OpBuilderWithAsyncTaskIds &builder, ttng::TCGen5MMAOp mmaOp,
   }
   phase = builder.createWithAsyncTaskIds<arith::ExtSIOp>(
       loc, builder.getI32Type(), phase);
-  return builder.createWithAsyncTaskIds<ttng::WaitBarrierOp>(
+  auto waitOp = builder.createWithAsyncTaskIds<ttng::WaitBarrierOp>(
```

Comment:
This impacts one barrier that otherwise would have introduced a correctness issue.